src/Tizen.System.Session/Session/Session.cs

```diff
@@ -0,0 +1,509 @@
+        /// <exception cref="InvalidOperationException">Provided subsession user ID does not exist</exception>
+        /// <exception cref="OutOfMemoryException">Out of memory</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public Task SubsessionRemoveUserAsync(string userName)
+        {
+            var task = new TaskCompletionSource<bool>();
+            int taskID = 0;
+
+            lock (_replyLock)
+            {
+                taskID = _replyID++;
+            }
+
+            _replyMap[taskID] = (int result, IntPtr data) =>
+            {
+                _replyMap.Remove((int)data);
+
+                try
+                {
+                    CheckError((SessionError)result, "Interop failed to remove a subsession user");
+                }
+                catch (Exception exception)
+                {
+                    task.SetException(exception);
+                    return;
+                }
+
+                task.SetResult(true);
+            };
+
+            SessionError ret = Interop.Session.SubsessionRemoveUser(SessionUID, userName, _replyMap[taskID], (IntPtr)taskID);
+            CheckError(ret, "Interop failed to register a reply for removing a user");
+            return task.Task;
+        }
+
+        /// <summary>
+        /// Request a subession to become currently active.
+        /// </summary>
+        /// <param name="userName">Existing subesssion user ID to be set as active</param>
+        /// <remarks>
+        /// Subsession ID must not start with a dot or have slashes.
+        /// Special subsession ID "" (empty string, see EmptyUser) can be switched to, when it's required to deactivate
+        /// current subsession (this step is needed when current session is to be removed).
+        /// </remarks>
+        /// <exception cref="ArgumentException">Session UID of this object is invalid, or user ID is not a valid subession ID</exception>
+        /// <exception cref="InvalidOperationException">Provided subsession user ID to switch to does not exist</exception>
+        /// <exception cref="OutOfMemoryException">Out of memory</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public Task SubsessionSwitchUserAsync(string userName)
+        {
+            var task = new TaskCompletionSource<bool>();
+            int taskID = 0;
+
+            lock (_replyLock)
+            {
+                taskID = _replyID++;
+            }
+
+            _replyMap[taskID] = (int result, IntPtr data) =>
+            {
+                _replyMap.Remove((int)data);
+
+                try
+                {
+                    CheckError((SessionError)result, "Interop failed to switch to a different subsession user");
+                }
+                catch (Exception exception)
+                {
+                    task.SetException(exception);
+                    return;
+                }
+
+                task.SetResult(true);
+            };
```

Comment:
```suggestion
            _replyMap[taskID] = (int result, IntPtr data) =>
            {
                try
                {
                    CheckError((SessionError)result, "Interop failed to switch to a different subsession user");
                }
                catch (Exception exception)
                {
                    task.SetException(exception);
                    return;
                }

                task.SetResult(true);
                _replyMap.Remove((int)data);
            };
```
You should remove the callback at the last of callback. if not native part of callback was GCed and it can't return to native